File: app/src/main/java/com/xeniac/warrantyrostermanager/NetworkHelper.java
package com.xeniac.warrantyrostermanager;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

public class NetworkHelper {

    public static boolean hasNetworkAccess(Context context) {
        ConnectivityManager connectivityManager =
                (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);

        try {
            NetworkInfo activeNetwork = connectivityManager.getActiveNetworkInfo();
            return activeNetwork != null && activeNetwork.isConnectedOrConnecting();
        } catch (Exception e) {
            e.printStackTrace();
            return false;
        }
    }
}

File: app/src/main/java/com/xeniac/warrantyroster_manager/util/NetworkHelper.kt
package com.xeniac.warrantyroster_manager.util

import android.content.Context
import android.net.ConnectivityManager

class NetworkHelper {
    companion object {
        fun hasNetworkAccess(context: Context): Boolean {
            val connectivityManager = context
                .getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager

            return try {
                val activeNetwork = connectivityManager.activeNetworkInfo
                activeNetwork != null && activeNetwork.isConnectedOrConnecting
            } catch (e: Exception) {
                e.printStackTrace()
                false
            }
        }
    }
}